apps/dashboard/app/apps/featured_app.rb

```diff
@@ -1,21 +1,19 @@
-# A simple subclass of OodApp that overrides category
-# and subcategory to allow for groupings other than what's
+# A simple subclass of OodApp that provides 2 new properties feature
+# and sub_feature to allow for groupings other than what's
 # provided in the app's manifest & definition.
 class FeaturedApp < OodApp
-  alias_method :original_category, :category
-  alias_method :original_subcategory, :subcategory
 
-  attr_reader :category, :subcategory, :token
+  attr_reader :feature, :sub_feature, :token
 
-  def self.from_ood_app(app, token: nil)
-    FeaturedApp.new(app.router, token: token)
+  def self.from_pinned_app(app, token: nil)
+    FeaturedApp.new(app.router, token: token, feature: I18n.t("dashboard.pinned_apps_category"), sub_feature: I18n.t('dashboard.pinned_apps_title'))
   end
 
-  def initialize(router, category: I18n.t("dashboard.pinned_apps_category"), subcategory: I18n.t('dashboard.pinned_apps_title'), token: nil)
+  def initialize(router, token: nil, feature: nil, sub_feature: nil)
     super(router)
 
-    @category = category.to_s
-    @subcategory = subcategory.to_s
+    @feature = feature.to_s
+    @sub_feature = sub_feature.to_s
     @token = token || router.token
   end
 
```

Comment:
This ^^ is an idea to fix the `categories` / `original_categories` issue when creating `FeaturedApps`, as well as making it a bit more generic to be reused.

I am using `FeaturedApps` when creating a flat list of apps and sub apps for the token matcher.

At the moment the logic is in NavBar, but it makes sense to consolidate it with the token matcher logic in `Router`